### ISSUE TITLE
Setting the default network for the cluster

### ISSUE THREAD

**Issue by stevenctl** [istio/istio#46772]
We currently have two places for this, meaning two install steps. If one is applied but not the other, it may break things. 

* `values.global.network`. This value is used during injection to set `ISTIO_META_NETWORK` and `topology.istio.io/network` on Pods. 
* `topology.istio.io/network` on the system namespace. This is used to infer the network for Pods that don't have the label (aren't injected). 

This could cause:
* A zTunnel having `ISTIO_META_NETWORK="foo"` but it's captured pods have `network=""`. Network+IP is still our key for lookups (unless we resolve #46628). I don't think we have `ISTIO_META_NETWORK` in the daemonset template today though. Need to solve this eventually. 
* A sidecar proxy will attempt to send cross network to pods that can't receive traffic through the eastwest gateway. 

One possible solution (open to  other ideas): Make these things consistent.

* If the namespace label isn't set, fallback to `values.global.network`. This works for the _local_ cluster, but not remote clusters. 
* If the label is set on the local cluster, use that for injection instead Possibly deprecate `values.global.network`?

**Comment by stevenctl** [comment]
I intentionally left out https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshNetworks. A lot of what was written above was to make the config something that could be discovered across clusters and not a configmap that needed to be replciated and kept in sync. 

That is a valid option for some users, but not all.

**Comment by bleggett** [comment]
Status quo effectively means `values.global.network` doesn't do anything for Ambient, since we don't label/inject those pods, and we also don't use it to label `istio-system`

**Comment by stevenctl** [comment]
Well.. I happened to make it do something very recently: https://github.com/istio/istio/pull/46725

**Comment by stevenctl** [comment]
> why would this not working for remote cluster

We _must_ either have a mapping in MeshNetworks of cluster name  -> network name, or read the label from the remote cluster's system namespaces. 

Istiod in cluster 1 with values.global.network=network-1 is reading Pods from cluster-2, with a values.global.network of network-2. It needs to know the default network is not the same. If the network is the same, we can use it's endpoints on Pods regardless of whether they have a proxy.

**Comment by linsun** [comment]
> > why would this not working for remote cluster
> 
> We _must_ either have a mapping in MeshNetworks of cluster name -> network name, or read the label from the remote cluster's system namespaces.
> 
> Istiod in cluster 1 with values.global.network=network-1 is reading Pods from cluster-2, with a values.global.network of network-2. It needs to know the default network is not the same. If the network is the same, we can use it's endpoints on Pods regardless of whether they have a proxy.

Thanks!  I had forgotten we also support primary-remote for non flat networks.  Makes sense.

**Comment by hzxuzhonghu** [comment]
I think `ISTIO_META_NETWORK` should have higher priority over namespace network.  Think about the case of multi network per cluster.

**Comment by stevenctl** [comment]
Agreed. I think that is fine as is. For ambient, it may be better _not_ to set that unless it needs to override the cluster-wide default. 

For our various options that _do_ apply globally, they need to be consistent:
* MeshConfig/MeshNetworks
* Injector/operator value
* Label on system namespace

**Comment by hanxiaop** [comment]
not stale

**Comment by hanxiaop** [comment]
not stale

**Comment by chaospuppy** [comment]
> Status quo effectively means values.global.network doesn't do anything very useful for Ambient (except possibly break things), since by design we don't label/inject those pods, and we also don't use it to label istio-system
 
@bleggett is this still the case today?  I'm struggling to get a pretty simple setup going (#59584).  The [docs](https://istio.io/latest/docs/setup/install/virtual-machine/) still suggest that `global.network` must be set, and later that `topology.istio.io/network` must also be set on the system namespace.
